src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,41 @@
+import ModalCloseIcon from "@/icons/ModalCloseIcon";
+import { useEffect } from "react";
+
+type ModalProps = {
+  isModalOpen: boolean;
+  setIsModalOpen: (isOpen: boolean) => void;
+  children: React.ReactNode;
+};
+
+const Modal = ({ children, isModalOpen, setIsModalOpen }: ModalProps) => {
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === "Escape") {
+        setIsModalOpen(false);
+      }
+    };
+    window.addEventListener("keydown", handleKeyDown);
+    return () => window.removeEventListener("keydown", handleKeyDown);
+  }, [setIsModalOpen]);
+
+  return (
+    isModalOpen && (
+      <div
```

Comment:
To make the modal accessible, include ARIA attributes like `role="dialog"` and `aria-modal="true"`.